parsers/arangodb/arangodb.go

```diff
@@ -0,0 +1,196 @@
+	var v = make(map[string]interface{})
+	err = errors.New("Line is not a request log line.")
+	var abort bool
+	var s string
+
+	v[timestampFieldName], abort = firstWord(&line)
+	if abort { return }
+
+	s, abort = firstWord(&line)
+	if abort { return }
+	v[pidFieldName] = removeBrackets(s)
+
+	v[logLevelFieldName], abort = firstWord(&line)
+	if abort { return }
+
+  s, abort = firstWord(&line)
+	if abort { return }
+	v[logTopicFieldName] = s
+
+	if s != "{requests}" { return }
+
+	var fields []string = strings.Split(line, ",")
+	if v[logLevelFieldName] == "DEBUG" {
+		if len(fields) != 6 { return }
+		v[idFieldName] = removeQuotes(fields[1])
+		v[sourceIPFieldName] = removeQuotes(fields[2])
+		v[methodFieldName] = removeQuotes(fields[3])
+		v[protocolFieldName] = removeQuotes(fields[4])
+		v[fullURLFieldName] = removeQuotes(fields[5])
+	} else {
+		if len(fields) != 10 { return }
+		v[idFieldName] = removeQuotes(fields[1])
+		v[sourceIPFieldName] = removeQuotes(fields[2])
+		v[methodFieldName] = removeQuotes(fields[3])
+		v[protocolFieldName] = removeQuotes(fields[4])
+		v[resCodeFieldName], _ = strconv.ParseInt(fields[5], 10, 32)
+		v[reqBodyLenFieldName], _ = strconv.ParseInt(fields[6], 10, 64)
+		v[resBodyLenFieldName], _ = strconv.ParseInt(fields[7], 10, 64)
+		v[fullURLFieldName] = removeQuotes(fields[8])
+		v[totalTimeFieldName], _ = strconv.ParseFloat(fields[9], 64)
+  }
+  return v, nil
+}
+
+func (p *Parser) Init(options interface{}) error {
+	p.conf = *options.(*Options)
+	p.lineParser = &ArangoLineParser{}
+	return nil
+}
+
+func (p *Parser) ProcessLines(lines <-chan string, send chan<- event.Event) {
+	for line := range lines {
+		values, err := p.lineParser.ParseLogLine(line)
+		// we get a bunch of errors from the parser on ArangoDB logs, skip em
+		if err == nil {
+			timestamp, err := p.parseTimestamp(values)
+			if err != nil {
+				logFailure(line, err, "couldn't parse logline timestamp, skipping")
+				continue
+			}
+
+			logrus.WithFields(logrus.Fields{
+				"line":   line,
+				"values": values,
+			}).Debug("Successfully parsed line")
+
+			// we'll be putting the timestamp in the Event
+			// itself, no need to also have it in the Data
+			delete(values, timestampFieldName)
+
+			send <- event.Event{
+				Timestamp: timestamp,
+				Data:      values,
+			}
+		} else {
+			logFailure(line, err, "logline didn't parse, skipping.")
+		}
+	}
+	logrus.Debug("lines channel is closed, ending arangodb processor")
+}
+
+func (p *Parser) parseTimestamp(values map[string]interface{}) (time.Time, error) {
+	timestamp_value, ok := values[timestampFieldName].(string)
+	if ok {
+		var err error
+		for _, f := range timestampFormats {
+			var timestamp time.Time
+			timestamp, err = time.Parse(f, timestamp_value)
+			if err == nil {
+				return timestamp, nil
+			}
+		}
+		return time.Time{}, err
+	}
+
+	return time.Time{}, errors.New("timestamp missing from logline")
+}
+
+func logFailure(line string, err error, msg string) {
```

Comment:
It's intentional that only request lines are going to be parsed - all the other line types aren't errors, just omitted. Let's log them as INFO or DEBUG instead of ERROR level.